File: app/src/main/java/tech/pauly/findapet/discover/AnimalImageViewModel.java
package tech.pauly.findapet.discover;

import android.databinding.ObservableField;

import tech.pauly.findapet.data.models.Photo;
import tech.pauly.findapet.shared.BaseViewModel;

public class AnimalImageViewModel extends BaseViewModel {
    public ObservableField<String> imageUrl = new ObservableField<>("");

    public AnimalImageViewModel(Photo photo) {
        imageUrl.set(photo.getUrl());
    }
}

File: app/src/main/java/tech/pauly/findapet/discover/AnimalImageViewModel.kt
package tech.pauly.findapet.discover

import android.databinding.ObservableField
import tech.pauly.findapet.data.models.AnimalUrl

import tech.pauly.findapet.shared.BaseViewModel

class AnimalImageViewModel(url: AnimalUrl) : BaseViewModel() {

    var imageUrl = ObservableField<AnimalUrl>()

    init {
        imageUrl.set(url)
    }
}
